src/main/java/com/ixortalk/autoconfigure/oauth2/auth0/mgmt/api/Auth0ManagementAPI.java

```diff
@@ -74,15 +77,32 @@ public Auth0ManagementAPI(String domain, String createUserConnection, Authorized
     }
 
     @Cacheable(cacheNames = AUTH_0_USER_CACHE, sync = true)
-    public Map<String, User> listUsersByEmail() {
+    public Optional<User> getUserById(String userId) {
         try {
-            return
-                    listItemsFromAllPages(getManagementAPI().users()::list, new UserFilterAdapter())
-                            .stream()
-                            .collect(toMap(User::getEmail, identity()));
+            return Optional.ofNullable(getManagementAPI().users().get(userId, new UserFilter())
+                    .execute());
+        } catch (Auth0Exception e) {
+            LOGGER.error("Error retrieving user: " + e.getMessage(), e);
+            throw new Auth0RuntimeException("Error retrieving user: " + e.getMessage(), e);
+        }
+    }
+
+    @Cacheable(cacheNames = AUTH_0_USER_CACHE, sync = true)
```

Comment:
We are now using one (`AUTH_0_USER_CACHE `) cache where the same user can exist for different keys.  In theory a `userId` will never be equal to the `userEmail`, but I don't think it's a good idea to do this?